packages/pages-shared/src/asset-server/rulesEngine.ts

```diff
@@ -83,17 +83,15 @@ export const generateRulesMatcher = <T>(
 	][];
 
 	return ({ request }: { request: Request }) => {
-		const { pathname, host, search } = new URL(request.url);
-		const sortedSearch =
-			search && search.length
-				? "?" + search.slice(1).split("&").sort().join("&")
-				: search;
+		const { pathname, host, search, searchParams } = new URL(request.url);
+		searchParams.sort()
```

Comment:
This line returns a new array, it doesn't modify the existing one

Reply:
Really? https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/sort

Reply:
https://cloudflareworkers.com/#ae92b73ec9000e32b2d0ab69218f21af:https://tutorial.cloudflareworkers.com/greg?z&y&x